lib/fourOhFour.js

```diff
@@ -33,7 +33,7 @@ function fourOhFour (options) {
   const { logger, genReqId } = options
 
   // 404 router, used for handling encapsulated 404 handlers
-  const router = FindMyWay({ onBadUrl: createOnBadUrl(), defaultRoute: fourOhFourFallBack })
+  const router = FindMyWay({ constraints: options.constraints, onBadUrl: createOnBadUrl(), defaultRoute: fourOhFourFallBack })
```

Comment:
We introduced this to fix a failing test

Reply:
Which test is failing? Because I can only see the CI fail because of `npm`.

Reply:
I'm not at the Mac at the moment.
Execute `npm i --force` and then run the test suite. It happened consistently for me.

Reply:
Your example fails (`console.log('called')` is reached) even without this change:

```js
'use strict'

const fastify = require('../fastify')({
  logger: false
})

fastify.setNotFoundHandler({
  constraints: { host: 'example.com' }
}, function (request, reply) {
  console.log('called')
  reply.send(404)
})

fastify
  .get('/', function (req, reply) {
    reply
      .send({ hello: 'world' })
  })

fastify.listen({ port: 3000 }, (err, address) => {
  if (err) throw err
})
```

Also, this was changed because CI was failing due to a new FindMyWay update I guess. I'll update the code removing it just to you see in the CI

Reply:
See: https://github.com/fastify/fastify/runs/6131465332?check_suite_focus=true